WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -928,6 +930,57 @@ private extension AuthenticationManager {
         }
         navigationController.show(errorViewController, sender: self)
     }
+
+    func checkWooInstallation(in navigationController: UINavigationController,
+                              onSuccess: @escaping () -> Void) {
+        let action = SitePluginAction.getPluginDetails(siteID: WooConstants.placeholderStoreID, pluginName: Constants.wooPluginName) { result in
```

Comment:
@itsmeichigo I was just taking a look to compare how the implementation differs between the two platforms, and if I'm understanding things correctly, you are using the endpoint `/wp/v2/plugins` to check if Woo is installed here, I don't think this can work for all cases, users with the role "shop manager" can't use this endpoint.

As detailed in the internal doc pe5sF9-U3-p2, there are two ways to check if Woo is installed for all the users. Currently, in Android [we are re-using](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2614) the same check as Jetpack CP sites (meaning `/wc/v3/settings`), but if we decide to remove XMLRPC, we might try to move to use the root endpoint `/wp-json/` in order to reduce the number of requests we send.

Reply:
Thanks for looking into my code @hichamboushaba! Could you clarify how we can use the root endpoint to check for Woo please? And how does it have anything to do with XMLRPC?

Reply:
> Could you clarify how we can use the root endpoint to check for Woo please? 

This would be the same as we do for checking the Woo API version ([here](https://github.com/woocommerce/woocommerce-ios/blob/9cfd35f8f2f70f1d48353b9b950012cf609989aa/WooCommerce/Classes/Tools/RequirementsChecker.swift#L79-L97)), the root endpoint returns an array of the available namespaces of APIs in the site, and if it contains `/wc/{something}`, we can infer that the site has WooCommerce.

> And how does it have anything to do with XMLRPC?

Sorry for not being clear about this, this is specific to Android, currently the site data is being fetched using XMLRPC, so as long as we don't touch this, it made sense to me to reuse the same WooCommerce check that we use for Jetpack CP sites. But if we decide to remove XMLRPC, we will use the root endpoint to fetch site data, so in order to save one request, I would use this same request to infer if Woo is installed or not too, and stop using `/wc/v3/settings`.

Reply:
Thanks for the detailed explanation Hicham! I'll use the root endpoint to check too, for simplicity 🙇